maintainers/scripts/instantiate-constitutents.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Dumps a Makefile which can be used to instantiate the derivations that Hydra will evaluate
+
+# Usage:
+#  $1   = nix expression to `import`, such as <nixpkgs/nixos/release-small.nix>
+#  $2   = attribute within $1, which should be a Hydra job attrset (i.e. has a .constituents attribute)
+#  $3   = additional arguments for nix-instantiate, such as 'system = "aarch64-linux";'
+
+# Example:
+#   make -j32 -f <(instantiate-constitutents.sh '<nixpkgs/nixos/release-small.nix>' tested 'supportedSystems = ["x86_64-linux"];')
+
+DEST="$(mktemp -d)"
+echo "drv files will be written to $DEST" 1>&2
+IMPORT="$1"
+shift
+ATTR="$1"
+shift
+ARGS="$1"
+
+echo "default: build"
+for ATTR in $(nix eval --raw --impure --expr "let lib = import <nixpkgs/lib>; in lib.concatStringsSep \"\\n\" (with import $IMPORT {$ARGS}; $ATTR.constituents)")
```

Comment:
Move the eval outside the `for ... in` and into a variable

Reply:
2002a491c657